src/sdam/description/topology/server_selection/test/in_window.rs

```diff
@@ -211,39 +216,54 @@ async fn load_balancing_test() {
     let mut handler = EventHandler::new();
     let mut subscriber = handler.subscribe();
     let mut options = CLIENT_OPTIONS.clone();
+    let max_pool_size = 10;
+    let hosts = options.hosts.clone();
     options.local_threshold = Duration::from_secs(30).into();
+    options.max_pool_size = Some(max_pool_size);
+    options.min_pool_size = Some(max_pool_size);
     let client = TestClient::with_handler(Some(Arc::new(handler.clone())), options).await;
 
-    // wait for both servers to be discovered.
-    subscriber
-        .wait_for_event(Duration::from_secs(30), |event| {
-            if let Event::Sdam(SdamEvent::TopologyDescriptionChanged(event)) = event {
-                event
-                    .new_description
-                    .servers()
-                    .into_iter()
-                    .filter(|s| matches!(s.1.server_type(), ServerType::Mongos))
-                    .count()
-                    == 2
-            } else {
-                false
-            }
-        })
-        .await
-        .expect("timed out waiting for both mongoses to be discovered");
+    // wait for both servers pools to be saturated.
+    for address in hosts {
+        let selector = Arc::new(move |sd: &ServerInfo| sd.address() == &address);
+        for _ in 0..max_pool_size {
+            let client = client.clone();
+            let selector = selector.clone();
+            runtime::execute(async move {
+                let options = FindOptions::builder()
+                    .selection_criteria(SelectionCriteria::Predicate(selector))
+                    .build();
+                client
+                    .database("load_balancing_test")
+                    .collection::<Document>("load_balancing_test")
+                    .find(doc! { "$where": "sleep(500) && true" }, options)
```

Comment:
why do we use the sleep(500)?

Reply:
This was done to ensure the connections weren't returned to the pools immediately and reused. I think this could've been achieved with failpoints and blockConnection too, but this was a bit easier.